packages/preview-middleware/templates/test/qunit.js

```diff
@@ -1,8 +1,4 @@
-sap.ui.define([
-    'sap/ui/thirdparty/qunit-2'
-], function(QUnit) {
-        QUnit.config.autostart = false;
-        sap.ui.require(<%- JSON.stringify(tests) %>, function() {
-            QUnit.start();
-    });
-});
\ No newline at end of file
+QUnit.config.autostart = false;
```

Comment:
[UI5 recommendation](https://github.com/SAP/open-ux-tools/pull/1813#discussion_r1553301524) was not to go for the global QUnit. Is this change really needed?

Reply:
@petermuessig I need your help here :) I also preferred the other approach but it wasn't working in karma for me.

Reply:
The issue seems to be that qunit starts despite `QUnit.config.autostart = false;` (I don't know why 🤷🏻)
Also this issue seems to only occur when proxying the ui5 resources and not when serving them via tooling from npmjs.

According to [Object preconfig chapter](https://qunitjs.com/api/config/) in the Qunit documentation (no deep link available, sorry)

_You can create a QUnit global variable with a config property, before QUnit itself is loaded. Any properties on this QUnit.config placeholder object will be carried over and applied to the real QUnit.config object once it exists._

So whereas `QUnit.config.autostart = false;` did not work as preconfig in a `script` tag of the html file (again, I don't know why 🤷🏻), the following did actually work when put even before the ui5 bootstrap script tag:
`window.QUnit = { config: {autostart: false}};`

Weirdly also `window.QUnit.config.autostart = false` did not work 🤷🏻

With this setup `npm run test` works in the playground but karma is quitting with an error message

```
Chrome Headless 126.0.6478.115 (Windows 10) ERROR
  Uncaught TypeError: d.done is not a function
  at <path_to_playground>/node_modules/karma-ui5/dist/browser-bundle.js:15:458

  TypeError: d.done is not a function
      at d (<path_to_playground>/node_modules/karma-ui5/dist/browser-bundle.js:2:95253)
      at l.useIframe.g._frame.onload (<path_to_playground>/node_modules/karma-ui5/dist/browser-bundle.js:2:96262)
``` 

Also `QUnit.config.autostart = false;` seems to be a fair approach at least according to the [qunit documentation](https://qunitjs.com/api/config/autostart/)

Reply:
UPDATE: `karma-ui5` mistakes the `window.Qunit` preconfig for the real `QUnit`. That's why karma now quits with an error message. `d.done` actually is `Qunit.done` which is indeed not a function in the precondition object.
So the question to me is, if that's actually a bug in `ui5-karma` 🤔

Reply:
@heimwege please check https://github.com/SAP/open-ux-tools/pull/1998/commits/596ba41e7304637d5f1cd458f87437426f4d6d94 . It tried setting the window object before loading QUnit based on your feedback and that seems to work for me with karma as well as when using the ui5-tooling

Reply:
So the actual issue seems to be caused by the `data-sap-ui-oninit`. Using the required `QUnit` works perfectly fine as long as the `qunit.js` is not loaded via `data-sap-ui-oninit` but using a separate script tag in the `qunit.html`. So the changes in the .js file seem to be not needed.

Reply:
Just to be sure, I get it correct, we can remove
```
window.QUnit ??= {};
QUnit.config ??= {};
QUnit.config.autostart = false;
```
again? 

The key change was that we move the qunit.js file into a script tag instead of an onInit script?

Reply:
At least that's what worked for me 🤷🏻 I would of course still be interested in the opinions of @petermuessig and @matz3.

Reply:
I have updated the `qunit.js` template in https://github.com/SAP/open-ux-tools/pull/1998/commits/1cf3e0605dd65a1d59b71e26deea91ffddac80ba one more time based on the feedback by @petermuessig.
@heimwege could you please do yet another test run in your repo? For me it works now with `karma`, `ui5-test-runner` and `ui5-tooling`.